### ISSUE TITLE
Installing Website App Fails in V16 Community Edition

### ISSUE THREAD

**Issue by jimbiggins** [odoo/odoo#103555]
Impacted versions: 16 (nightly build 16.0.20221019)
App: Website
Hosting environment: AWS Lightsail with 1CPU and 2GB Ram.
OS: Ubuntu 20.04 LTS
Postgres: 12.12
Odoo Installed from: https://nightly.odoo.com/16.0/nightly/deb/odoo_16.0.20221019.tar.xz using apt-get  using instructions located at https://www.odoo.com/documentation/16.0/administration/install/install.html#setup-install-packaged

Steps to reproduce: Create a new DB via web UI, log in, Install "Website" app, fill in Website creation wizard.  After waiting about two minutes at "creating your website" animation, returns "Internal Server Error".

### Actual error output:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 1534, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 134, in retrying
    result = func()
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 1563, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 1676, in dispatch
    return self.request.registry['ir.http']._dispatch(endpoint)
  File "/usr/lib/python3/dist-packages/odoo/addons/website/models/ir_http.py", line 222, in _dispatch
    response = super()._dispatch(endpoint)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_http.py", line 138, in _dispatch
    result = endpoint(**request.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 673, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/usr/lib/python3/dist-packages/odoo/addons/website/controllers/main.py", line 98, in index
    website_page = request.env['ir.http']._serve_page()
  File "/usr/lib/python3/dist-packages/odoo/addons/website/models/ir_http.py", line 260, in _serve_page
    page = request.env['website.page'].sudo().search(published_domain, order='website_id asc', limit=1)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 1536, in search
    res = self._search(domain, offset=offset, limit=limit, order=order, count=count)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 4618, in _search
    query = self._where_calc(domain)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 4386, in _where_calc
    return expression.expression(domain, self).query
  File "/usr/lib/python3/dist-packages/odoo/osv/expression.py", line 447, in __init__
    self.parse()
  File "/usr/lib/python3/dist-packages/odoo/osv/expression.py", line 675, in parse
    parent_model = model.env[field.related_field.model_name]
AttributeError: 'NoneType' object has no attribute 'model_name'

**Comment by jimbiggins** [comment]
Found the solution to this.  I had to add the following to my nginx configuration inside the server{} config:

`add_header Content-Security-Policy upgrade-insecure-requests;`

I have nginx doing redirecting http -> https and handling SSL.  In this configuration, Odoo never correctly auto-detects the base url and I have to change it.  In this case changing base url did not fix this issue but adding the nginx setting did.
